### PR TITLE
cache parquet, add optional warning when variables are missing

### PR DESCRIPTION
**Pull request by TomGoBravo**
This is part of https://trello.com/c/7HrFDkv7/773-import-ccd20-state-county-dashboard-data-into-combineddatasets

This change
* Adds a cached copy of the parquet file in git LFS so scripts that depend on it can be run deterministically
* Adds click to the 3 scripts using ccd_helpers with --fetch/--no-fetch command line flag
* Adds optional warning when a variable in the upstream file is not in the variables list

These changes make it easier to add and modify scripts using ccd_helpers. The scripts finish faster and warnings help me find and fix typos.

### CODE REVIEW

**Review by ghop02 — COMMENTED** [review]
Overall looks good!  

The inclusion of all the variables seems fine for now (and helpful for further work), I wonder if there's some way that we could expose the list of variables from a future endpoint. Maybe something that looks at https://github.com/covid-projections/can-scrapers/blob/main/can_tools/bootstrap_data/covid_variables.csv

File: scripts/ccd_helpers.py [hunk 1]
```diff
@@ -122,11 +144,28 @@ def query_multiple_variables(self, variables: List[ScraperVariable]) -> pd.DataF
...
+                _logger.info(
+                    "Upstream has variable not in variables list",
+                    variable_name=variable_name,
+                    count=count,
+                )
+
     @staticmethod
-    def load_from_url(url: str = DATA_URL) -> "CovidCountyDataset":
-        """Loads CovidCountyData from specficied url, performing minor cleanup."""
+    def load(*, fetch: bool) -> "CovidCountyDataset":
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/ccd_helpers.py:155]: I prefer that function APIs try to only expose what is run in production or at least tested. In this case I'm expecting all production calls to `load` to include a bool taken from the command line so a default value is not used or tested.

File: scripts/update_can_scraper_state_providers.py [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 from covidactnow.datapublic import census_data_helpers
 from covidactnow.datapublic.common_fields import CommonFields
+from covidactnow.datapublic.common_fields import FieldNameAndCommonField
+from covidactnow.datapublic.common_fields import GetByValueMixin
 
 from scripts import helpers
 from scripts import ccd_helpers
 
+
+pd.set_option("mode.chained_assignment", "raise")
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/update_can_scraper_state_providers.py:19]: Good call. Done.

File: scripts/ccd_helpers.py [hunk 3]
```diff
@@ -39,64 +46,73 @@ class Fields(GetByValueMixin, FieldNameAndCommonField, enum.Enum):
...
+    """Represents a specific variable scraped in CAN Scraper Dataset.
+
+    If `common_field` is None the scraper variable is dropped instead of returned.
+    """
 
     variable_name: str
-    measurement: str
     provider: str
-    unit: str
+    measurement: str = ""
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/ccd_helpers.py:56]: Added some more checks and comments.

File: scripts/ccd_helpers.py [hunk 4]
```diff
@@ -39,64 +46,73 @@ class Fields(GetByValueMixin, FieldNameAndCommonField, enum.Enum):
...
+        return all_df.loc[is_selected_data, :].copy()
 
-    def query_multiple_variables(self, variables: List[ScraperVariable]) -> pd.DataFrame:
+    def query_multiple_variables(
+        self, variables: List[ScraperVariable], *, log_provider_coverage_warnings: bool = False
+    ) -> pd.DataFrame:
         """Queries multiple variables returning wide df with variable names as columns.
 
         Args:
             variable_queries: Variables to query
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/ccd_helpers.py:94]: Done

File: scripts/ccd_helpers.py [hunk 5]
```diff
@@ -39,64 +46,73 @@ class Fields(GetByValueMixin, FieldNameAndCommonField, enum.Enum):
...
-                & (all_df[Fields.RACE] == variable.race)
-                & (all_df[Fields.SEX] == variable.sex)
-            )
-            if variable.unit:
-                is_selected_data = is_selected_data & (all_df[Fields.UNIT] == variable.unit)
-
-            data = all_df.loc[is_selected_data]
+            if variable.common_field is None:
+                assert variable.measurement == ""
+                assert variable.unit == ""
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/ccd_helpers.py:103]: Added comment

File: scripts/update_can_scraper_state_providers.py [hunk 6]
```diff
@@ -19,8 +27,39 @@
 
 
 def transform(dataset: ccd_helpers.CovidCountyDataset):
-
     variables = [
+        ccd_helpers.ScraperVariable(variable_name="pcr_tests_negative", provider="state"),
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply scripts/update_can_scraper_state_providers.py:31]: I had a similar thought but decided what I had was good enough and not to distract myself futzing around with a class hierarchy.

File: update.sh [hunk 7]
```diff
@@ -19,7 +24,6 @@ python scripts/update_covid_county_data.py
 # python scripts/update_aws_lake.py --replace-local-mirror --cleanup-local-mirror
 python scripts/update_hhs_testing_data.py
 python scripts/update_hhs_hospital_data.py
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply update.sh:26]: Added TODO

**Review by ghop02 — APPROVED** [review]
LGTM!

**Pull request merged**